org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/semanticTokens/SemanticHighlightReconcilerStrategy.java

```diff
@@ -171,7 +174,7 @@ private Function<Position, Integer> offsetMapper() {
 		};
 	}
 
-	private SemanticTokensParams getSemanticTokensParams() {
+	private static SemanticTokensParams getSemanticTokensParams(IDocument document) {
```

Comment:
I see that the `document` is still a class field, even though many methods are now static and take it as an argument. Should we maybe split the class so that we do not have methods which take a variable as an argument that hides a field in the class?

Reply:
I see, let me look into this...